### PR TITLE
First real commit

### PR DESCRIPTION
**Pull request by xamchuk**

### CODE REVIEW

File: test5_json/NetworkManager.swift [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        dataTask.resume()
+    }
+    
+    func loadImage(url: URL, completion: @escaping (Data) -> Void) {
+        let dataTask = URLSession.shared.dataTask(with: url) { (data, response, error) in
+            
+            if let err = error {
+                print("failed to get Users Avatar\(err)")
+            }
+            do {
```
Comment (Alex-Alex77) [inline test5_json/NetworkManager.swift]: You don't need do {} block here, because there is no try inside

File: test5_json/NetworkManager.swift [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                guard let dataImage = data else { return }
+                DispatchQueue.main.async {
+                     completion(dataImage)
```
Comment (Alex-Alex77) [inline test5_json/NetworkManager.swift]: We load UIImage and not Data, so we need to return UIImage

File: test5_json/MainViewController.swift [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        if filteredRepositories.count > 0 {
+            return filteredRepositories.count
+        } else {
+        return items.count
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath) as? RepositoriesUserTableViewCell else { return UITableViewCell() }
+        var avatarUrl = items[indexPath.row].owner.avatarUrl {
```
Comment (Alex-Alex77) [inline test5_json/MainViewController.swift]: This variable doesn't belong here. Move it inside the cell

Comment (Alex-Alex77) [reply test5_json/MainViewController.swift]: First time that I see didSet block inside a function. It's possible, but definitely not good code

**Comment by Alex-Alex77** [comment]
@xamchuk Make changes, then create new commit with it. Then push your local branch one more time

File: test5_json/MainViewController.swift [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+            didSet {
+                networkManager.loadImage(url:avatarUrl ) { (dataImage) in
+                    let image = UIImage(data: dataImage)
+                    cell.imageView1.image = image
+                    tableView.reloadData()
```
Comment (Alex-Alex77) [inline test5_json/MainViewController.swift]: That's a mistake. You can't reload from cellForRowAtIndexPath. This creates infinite loop

File: test5_json/NetworkManager.swift [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+                
+            } catch let jsonError {
+                print("Failed to decode", jsonError)
+            }
+            
+        }
+        dataTask.resume()
+    }
+    
+    func loadImage(url: URL, completion: @escaping (Data) -> Void) {
```
Comment (Alex-Alex77) [inline test5_json/NetworkManager.swift]: here you need func loadImage(url: URL, completion: @escaping (UIImage) -> Void)

File: test5_json/NetworkManager.swift (hunk 2 above)
Comment (xamchuk) [reply test5_json/NetworkManager.swift]: I cannot write any UI (UIImage) there.

Comment (Alex-Alex77) [reply test5_json/NetworkManager.swift]: Of course, you need to create UIImage here

Comment (Alex-Alex77) [reply test5_json/NetworkManager.swift]: Why cannot?

Comment (Alex-Alex77) [reply test5_json/NetworkManager.swift]: If you mean you cannot use any UI here, then just think about it as a model. Your request returns UIImage like image

File: test5_json/NetworkManager.swift [hunk 6]
```diff
@@ -41,12 +41,10 @@ class NetworkManager {
...
             }
-            do {
-                guard let dataImage = data else { return }
-                DispatchQueue.main.async {
-                     completion(dataImage)
-                    }
-                }
+            guard let dataImage = data else { return }
+            DispatchQueue.main.async {
+                completion(dataImage)
```
Comment (Alex-Alex77) [inline test5_json/NetworkManager.swift]: You still returning Data, but you need to return UIImage. Read carefully all my comments

File: test5_json/MainViewController.swift [hunk 7]
```diff
@@ -74,23 +73,13 @@ extension MainViewController: UITableViewDataSource {
...
-            }
-        }
         
-        let url = items[indexPath.row].owner.avatarUrl
-            do {
-                let data = try Data(contentsOf: url)
-                cell.imageView1.image = UIImage(data: data)
-            } catch let error{
-                print(error.localizedDescription)
+        let avatarUrl = items[indexPath.row].owner.avatarUrl
```
Comment (Alex-Alex77) [inline test5_json/MainViewController.swift]: Your **cell** should contain var avatarURL. Move avatarURL variable to cell, remove it from here! Inside cell, you observe avatarURL changes by adding didSet block. So networkManager.loadImage... should be called from there

**Pull request merged**